tmk_core/common/keyboard.c

```diff
@@ -64,20 +64,41 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 
 #ifdef MATRIX_HAS_GHOST
+static uint16_t matrix_ghost_check[MATRIX_ROWS];
```

Comment:
I'm sure the compiler does a few optimizations here, since it's all static information. Size before (I'm using arm) 33560, size after: 33608. Total added: 48 bytes. The extra logic adds a bit (pun intended). I'd like to see it on AVR, might be different.